google/cloud/aiplatform/featurestore/featurestore.py

```diff
@@ -1273,9 +1299,39 @@ def batch_serve_to_df(
                     frames.append(message.to_dataframe())
 
         finally:
-            bigquery_client.delete_dataset(
-                dataset=temp_bq_dataset.dataset_id,
-                delete_contents=True,
-            )
+            # clean up: if user didn't specify dataset, delete ephemeral dataset
+            if bq_dataset_id is None:
+                bigquery_client.delete_dataset(
+                    dataset=temp_bq_dataset.dataset_id,
+                    delete_contents=True,
+                )
+
+            # clean up: if user specified BigQuery dataset, delete ephemeral tables
+            else:
+                bigquery_client.delete_table(temp_bq_batch_serve_table_id)
+                bigquery_client.delete_table(temp_bq_read_instances_table_id)
 
         return pd.concat(frames, ignore_index=True) if frames else pd.DataFrame(frames)
+
+    def _get_ephemeral_bq_full_dataset_id(
+        self, featurestore_id: str, project_number: str
+    ) -> str:
+        temp_bq_dataset_name = f"temp_{featurestore_id}_{uuid.uuid4()}".replace(
+            "-", "_"
+        )
+
+        project_id = resource_manager_utils.get_project_id(
+            project_number=project_number,
+            credentials=self.credentials,
+        )
+
+        return f"{project_id}.{temp_bq_dataset_name}"[:1024]
+
+    def _create_ephemeral_bq_dataset(
+        self, bigquery_client: bigquery.Client, dataset_id: str
+    ) -> "bigquery.Dataset":
```

Comment:
please add doc strings to this method as well

Reply:
fixed in 7020556